### ISSUE TITLE
Give GitHub Actions access to datasets for testing

### ISSUE THREAD

**Issue by jacobbieker** [openclimatefix/nowcasting_dataset#124]
In #107, it came up that we can't run some of the dataset tests because GitHub actions does not have access to the data. Giving it access could be quite useful, as we can then run tests on every PR.

We could try with some actions like these maybe? https://github.com/google-github-actions/setup-gcloud but yeah, I might add a small .nc of satellite data for unittesting the subsetting.

_Originally posted by @jacobbieker in https://github.com/openclimatefix/nowcasting_dataset/pull/107#discussion_r709992001_

**Comment by JackKelly** [comment]
Sounds good!

Would data have to be shared publicly for GitHub actions to access the data?

(Whilst we'd love to share as much data as possible, some data we're not allowed to share - such as the PVOutput.org data)

**Comment by jacobbieker** [comment]
I'm not entirely sure, I don't believe so, we could give it some kind of access token or something for running. Although once some of the data is public, we can always use that instead too

**Comment by jacobbieker** [comment]
Yeah, you can get access credentials with https://github.com/google-github-actions/get-secretmanager-secrets and you are able to even upload to GCS with this https://github.com/google-github-actions/upload-cloud-storage so there should be a way to access the data

**Comment by JackKelly** [comment]
Nice!
